lib/tasks/reset_qa.rake

```diff
@@ -0,0 +1,37 @@
+desc 'Reset the database to a state with some standard providers and test applications. Preserves existing support users'
+task reset_qa: :environment do
+  raise 'You can’t do this anywhere but QA or development!' unless HostingEnvironment.qa? || Rails.env.development?
+
+  Rake::Task['_reset_qa'].invoke
+end
+
+task _reset_qa: %i[
+  environment
+  backup_support_users
+  truncate_qa_database
+  restore_support_users
+  sync_dev_providers_and_open_courses
+  generate_test_applications
+]
+
+task truncate_qa_database: :environment do
+  raise 'You can’t do this anywhere but QA or development!' unless HostingEnvironment.qa? || Rails.env.development?
+
+  ActiveRecord::Tasks::DatabaseTasks.truncate_all
+end
+
+task backup_support_users: :environment do
+  existing_users = SupportUser.pluck(:email_address, :dfe_sign_in_uid)
+  Redis.new.set('provider_users', JSON.generate(existing_users))
```

Comment:
> 👍 Using Redis to stash these is a great idea. Should we use `support_users` as the key rather than `provider_users`?

oops — yes!

> 
> A (fairly unlikely) potential issue with this is that the process could fail after clearing the DB and before `restore_support_users` runs. One way to mitigate that would be to skip the backup if the `support_users` table is empty.

Good idea